mfrc522/mfrc522.py

```diff
@@ -0,0 +1,206 @@
+        TxSelReg            = 0x16
+        RxSelReg            = 0x17
+        RxThresholdReg      = 0x18
+        DemodReg            = 0x19
+        Reserved11          = 0x1A
+        Reserved12          = 0x1B
+        MfTxReg             = 0x1C
+        MfRxReg             = 0x1D
+        Reserved14          = 0x1E
+        SerialSpeedReg      = 0x1F
+        Reserved20          = 0x20
+        CRCResultRegL       = 0x21
+        CRCResultRegH       = 0x22
+        Reserved21          = 0x23
+        ModWidthReg         = 0x24
+        Reserved22          = 0x25
+        RFCfgReg            = 0x26
+        GsNReg              = 0x27
+        CWGsPReg            = 0x28
+        ModGsPReg           = 0x29
+        TModeReg            = 0x2A
+        TPrescalerReg       = 0x2B
+        TReloadRegH         = 0x2C
+        TReloadRegL         = 0x2D
+        TCounterValueRegH   = 0x2E
+        TCounterValueRegL   = 0x2F
+        Reserved30          = 0x30
+        TestSel1Reg         = 0x31
+        TestSel2Reg         = 0x32
+        TestPinEnReg        = 0x33
+        TestPinValueReg     = 0x34
+        TestBusReg          = 0x35
+        AutoTestReg         = 0x36
+        VersionReg          = 0x37
+        AnalogTestReg       = 0x38
+        TestDAC1Reg         = 0x39
+        TestDAC2Reg         = 0x3A
+        TestADCReg          = 0x3B
+        Reserved31          = 0x3C
+        Reserved32          = 0x3D
+        Reserved33          = 0x3E
+        Reserved34          = 0x3F
+
+    def __init__(self, spi_dev):
+        """Initializes a MFRC522 module.
+
+        spi_dev should be an object representing a SPI interface to which
+        the Reader is connected. It should have the following methods:
+          * transfer(bytes):  Selects the slave, transfers bytes, unselects
+                              the slave and returns received bytes.
+          * hard_powerdown(): Pulls NRST signal of the reader LOW, thus powering
+                              it down
+          * reset():          Pushes NRST signal of the reader HIGH,
+                              thus resetting it (and exiting the hard_powerdown)
+                              If the NRST is already high, this function shall
+                              pull it LOW and then HIGH again.
+        """
+
+        self.spi = spi_dev
+
+        self.reset()
+
+        self.write_register(MFRC522.Registers.TModeReg,      0x8D)
+        self.write_register(MFRC522.Registers.TPrescalerReg, 0x3E)
+        self.write_register(MFRC522.Registers.TReloadRegH,   0x00)
+        self.write_register(MFRC522.Registers.TReloadRegL,   0x1E)
+        self.write_register(MFRC522.Registers.TxASKReg,      0x40)
+        self.write_register(MFRC522.Registers.ModeReg,       0x3D)
+
+        self.antenna_on()
+
+    def write_register(self, register, val):
+        if type(val) is int:
+            self.spi.transfer(bytes((register.value << 1, val)))
+        else:
```

Comment:
Ugly handling of the "overloading" (two completely different + duplicating code paths when it's doing the same thing). What I'd prefer:
```python
        if type(val) is int:
            val = bytes([val])
        <paste from L120 onwards>
```